### ISSUE TITLE
[RFE] Support ARGSIZE from user input in trace_exec

### ISSUE THREAD

**Issue by michaelkatch** [inspektor-gadget/inspektor-gadget#3582]
## Current situation

Currently trace_exec is truncating arguments with length exceeding 128 bytes by default
![image](https://github.com/user-attachments/assets/61ede3eb-ebe6-4d64-9aea-2baf195bdf21)

## Impact
Some process executions are passed with longer parameters, specifically the first parameters (i.e. base64 encoded payloads) and we are unable to see them correctly

## Ideal future situation
I suggest accepting a parameter from the user indicating the maximum argument size we are willing to trace 
Ideally this argument will impact all of the arguments (The default maximum is 60) but increasing the length even for the first 10 arguments is reasonable

**Comment by alban** [comment]
Related issues:
- https://github.com/inspektor-gadget/inspektor-gadget/issues/2993
- https://github.com/inspektor-gadget/inspektor-gadget/issues/3491

I am not sure we can do it fully dynamic. Do your base64 payloads have a maximum size?

I see the maximum on Linux is 128KiB:
https://github.com/torvalds/linux/blob/v6.8/include/uapi/linux/binfmts.h#L9-L16

**Comment by michaelkatch** [comment]
@alban missed your comment , sorry about that.
After we discussed today, I suggested increasing the argument size to 256 bytes for the 5 arguments (that should be enough for us) and hopefully not take too much space

**Comment by alban** [comment]
tl;dr: we can even increase to 256 bytes for 20 arguments **and** reduce the size of the buffer.

---

There are 3 ebpf modules that trace exec arguments:
1. builtin trace-exec gadget
2. image-based trace_exec gadget
3. container-hook

Those 3 modules support up either 128 or 256 bytes for each argument:

```
$ git grep '#define ARGSIZE'
gadgets/trace_exec/program.bpf.c:18:#define ARGSIZE 128
pkg/container-hook/bpf/execruntime.h:5:#define ARGSIZE 256
pkg/gadgets/trace/exec/tracer/bpf/execsnoop.h:5:#define ARGSIZE 128
```

The container-hook package switched from 128 to 256 bytes in commit 99c369ef3554879053a669fb64b9173c3e01648b in order to support some setups with long paths (Docker on WSL2).

The 3 modules support up to 20 arguments. There is some left over code (coming from [bcc's execsnoop](https://github.com/iovisor/bcc/blob/8d85dcfac86bb7402a20bea5ceba373e5e019b6c/libbpf-tools/execsnoop.c#L75)) aiming to support 60 arguments but it has not been implemented in Inspektor Gadget. Nonetheless, the `struct event` is needlessly big enough to support 60 arguments.

So I suggest to change the argument size to 256 for the 20 arguments we support today. We can reduce the size of the args field in the `struct event` for the 20 arguments to `256*20 = 5 KiB` (instead of `128*60 = 7.5 KiB`).
